### PR TITLE
Replace recommonmark with MyST-Parser in docs

### PR DESCRIPTION
**Pull request by dgarcia360**
Replaces https://github.com/scylladb/scylla-operator/issues/1420

This pull request replaces recommonmark with MyST Parser. However, for backward compatibility, older versions will continue to use recommonmark.

## How to test

It could be a bit tricky:

1. Run the commands `make setup` and `make preview`. This will build the docs with Myst Parser.

2. In the `conf.py` file, add the master branch to the scylladb_markdown_recommonmark_versions list. Note: Once this pull request is merged, the master branch will use Myst-Parser.

    ```
    scylladb_markdown_recommonmark_versions = ['master', 'v1.8', 'v1.9', 'v1.10', 'v1.11']
    ```
3. Run the multiversion command [following these docs](https://sphinx-theme.scylladb.com/stable/configuration/multiversion.html#previewing-production-builds).  This will build the previous versions with recommonmark  - we don't have a version listed with MystParser yet.

4. Run the command `make multiversionpreview` and open the resulting URL. This will display the multiversion site.

### CODE REVIEW

**Comment by dgarcia360** [comment]
Added a new commit adding back the redirects.yaml file to fix the tests https://github.com/scylladb/scylla-operator/pull/1514/commits/068fe52a5536170775a68a85bdf86ef7cb7926d3

File: docs/source/_ext/scylladb_markdown.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import os
```
Comment (tnozicka) [inline docs/source/_ext/scylladb_markdown.py:1]: is this specific to our project or can this live in the theme?

**Comment by tnozicka** [comment]
I think we should make a tracking issue, to drop the old style and the workaround eventually (v1.14 master at earliest) and link it in the workarounds

**Comment by rzetelskik** [comment]
@dgarcia360 could you please squash the two of your commits and rename them to align with our convention (so basically drop the prefix and use a capital letter, e.g. `Add scylladb_markdown extension`)?

File: docs/source/_ext/scylladb_markdown.py (hunk 1 above)
Comment (dgarcia360) [reply docs/source/_ext/scylladb_markdown.py:1]: We can move it to the theme and distribute it as a new optional extension. However, I would prefer first to test against this project. Note that not all projects use MD to write their docs.

**Comment by dgarcia360** [comment]
@rzetelskik

> 
> could you please squash the two of your commits and rename them to align with our convention (so basically drop the prefix and use a capital letter, e.g. Add scylladb_markdown extension)?

Done!

**Comment by zimnx** [comment]
Several warnings are braking the build, seems like a space is missing before `note::` directive.
```
 /home/prow/go/src/github.com/scylladb/scylla-operator/docs/source/support/must-gather.md:27: WARNING: Pygments lexer name 'note::' is not known
/home/prow/go/src/github.com/scylladb/scylla-operator/docs/source/support/must-gather.md:66: WARNING: Pygments lexer name 'note::' is not known
/home/prow/go/src/github.com/scylladb/scylla-operator/docs/source/support/must-gather.md:89: WARNING: Pygments lexer name 'note::' is not known
/home/prow/go/src/github.com/scylladb/scylla-operator/docs/source/support/troubleshooting/installation.md:9: WARNING: Pygments lexer name 'note::' is not known
/home/prow/go/src/github.com/scylladb/scylla-operator/docs/source/support/troubleshooting/installation.md:14: WARNING: Pygments lexer name 'note::' is not known
/home/prow/go/src/github.com/scylladb/scylla-operator/docs/source/support/troubleshooting/installation.md:23: WARNING: Pygments lexer name 'note::' is not known 
```
This one triggers a warning:
https://github.com/scylladb/scylla-operator/blob/5820743fbe3276bb41a5b9e62ee54e0993f67818/docs/source/support/troubleshooting/installation.md?plain=1#L9
where this one doesn't:
https://github.com/scylladb/scylla-operator/blob/74c3b5e6f3ad1da38e84351268c8c81fdf33a4ca/docs/source/exposing.md?plain=1#L6

**Comment by zimnx** [comment]
New documentation pages were added recently and picked up upon rebasing. Several more migrations to new syntax are needed.

**Comment by rzetelskik** [comment]
thanks @zimnx 
@dgarcia360 I've updated my PR https://github.com/scylladb/scylla-operator/pull/1418, please rebase on top of it. That should get rid of the errors.

**Comment by rzetelskik** [comment]
Since we're in a bit of a hurry to fix the docs, @dgarcia360 let's set a deadline for rebasing this to this afternoon. If you won't have the cycles to do it today, I'll pick your changes and apply them to my PR https://github.com/scylladb/scylla-operator/pull/1418.

**Comment by rzetelskik** [comment]
Changes are being merged in #1418 . Closing in favor of the other PR.

**Comment by dgarcia360** [comment]
@rzetelskik Thanks!

**Pull request closed without merging**